packages/mui-material/src/IconButton/IconButton.test.js

```diff
@@ -34,18 +34,13 @@ describe('<IconButton />', () => {
     expect(container.querySelector('.touch-ripple')).not.to.equal(null);
   });
 
-  ['default', 'primary'].forEach((color) => {
-    it(`can disable the ripple and hover effect for color ${color}`, () => {
-      const { container, getByRole } = render(
-        <IconButton disableRipple color={color} TouchRippleProps={{ className: 'touch-ripple' }}>
-          book
-        </IconButton>,
-      );
-      expect(container.querySelector('.touch-ripple')).to.equal(null);
-      expect(getComputedStyle(getByRole('button'), ':hover').backgroundColor).to.equal(
-        getComputedStyle(getByRole('button')).backgroundColor,
-      );
-    });
+  it('can disable the ripple and hover effect', () => {
+    const { container } = render(
+      <IconButton disableRipple TouchRippleProps={{ className: 'touch-ripple' }}>
+        book
+      </IconButton>,
+    );
+    expect(container.querySelector('.touch-ripple')).to.equal(null);
```

Comment:
See https://github.com/mui/material-ui/pull/29298#discussion_r996201612 for why.